server/downloader.cpp

```diff
@@ -43,39 +45,36 @@ bool downloader::download(const std::string &name, const std::string &guid, uint
     return download_impl(relative_path);
 }
 
-std::string
-downloader::get_relative_path_str(const std::string &name, const std::string &guid, uint32_t age) {
+std::string downloader::get_relative_path_str(const std::string &name, const std::string &guid, uint32_t age) {
     std::stringstream ss;
     ss << std::hex << std::uppercase;
     ss << name << '/' << guid << age << '/' << name;
     return ss.str();
 }
 
-std::filesystem::path
-downloader::get_path(const std::string &name, const std::string &guid, uint32_t age) {
+std::filesystem::path downloader::get_path(const std::string &name, const std::string &guid, uint32_t age) {
     std::string relative_path = get_relative_path_str(name, guid, age);
-    auto path = std::filesystem::path(path_).append(relative_path);
+    auto path = std::filesystem::path(downloader::pdb_path_).append(relative_path);
     return path;
 }
 
 bool downloader::download_impl(const std::string &relative_path) {
-    std::lock_guard lock(mutex_);
```

Comment:
The lock is necessary here. Consider a situation where two people simultaneously initiate a request for the same pdb that is not cached locally, so that the pdb file is downloaded twice. But reading a cached pdb file does not require a lock, because there is no conflict in reading the same file concurrently.